src/gen_rules.ml

```diff
@@ -650,6 +650,72 @@ module Gen(P : Params) = struct
        Build.store_vfile vrequires);
     Build.vpath vrequires
 
+  let dot_merlin ~dir ~requires ~alias_modules =
+    (* Only generate .merlin file in default context *)
+    match Path.extract_build_context dir with
+    | Some ("default", remaindir) ->
+        let path = Path.relative remaindir ".merlin" in
+        add_rule (
+          requires
+          >>>
+          Build.arr (fun libs_list ->
+            libs_list
```

Comment:
You know, I was tempted to do this earlier, but I didn't think it would actually work.  To your credit, not only did it work, but it also worked without any changes downstream.